experiments/graph_oriented_layout_design/gold_cost_objectives.cpp

```diff
@@ -0,0 +1,144 @@
+
+    static constexpr const uint64_t bench_select = fiction_experiments::trindade16 | fiction_experiments::fontes18;
+
+    for (const auto& benchmark : fiction_experiments::all_benchmarks(bench_select))
+    {
+        for (uint64_t cost = 0; cost < 4; cost++)
+        {
+            if (cost == 0)
+            {
+                graph_oriented_layout_design_params.cost =
+                    fiction::graph_oriented_layout_design_params::cost_objective::AREA;
+            }
+            else if (cost == 1)
+            {
+                graph_oriented_layout_design_params.cost =
+                    fiction::graph_oriented_layout_design_params::cost_objective::WIRES;
+            }
+            else if (cost == 2)
+            {
+                graph_oriented_layout_design_params.cost =
+                    fiction::graph_oriented_layout_design_params::cost_objective::CROSSINGS;
+            }
+            else
+            {
+                graph_oriented_layout_design_params.cost =
+                    fiction::graph_oriented_layout_design_params::cost_objective::ACP;
+            }
+
+            auto network = read_ntk<fiction::tec_nt>(benchmark);
+
+            auto gate_level_layout = fiction::graph_oriented_layout_design<gate_lyt, fiction::tec_nt>(
+                network, graph_oriented_layout_design_params, &graph_oriented_layout_design_stats);
+
+            if (gate_level_layout.has_value())
+            {
+                // check equivalence
+                const auto eq_stats =
+                    fiction::equivalence_checking<fiction::technology_network, gate_lyt>(network, *gate_level_layout);
+
+                std::string eq_result = "NO";
+                if (eq_stats == fiction::eq_type::STRONG)
+                {
+                    eq_result = "STRONG";
+                }
+                else if (eq_stats == fiction::eq_type::WEAK)
+                {
+                    eq_result = "WEAK";
+                };
+                // calculate bounding box
+                const auto bounding_box = fiction::bounding_box_2d(*gate_level_layout);
+
+                const auto width  = bounding_box.get_x_size() + 1;
+                const auto height = bounding_box.get_y_size() + 1;
+                const auto area   = width * height;
+
+                const auto num_wires =
+                    gate_level_layout->num_wires() - gate_level_layout->num_pis() - gate_level_layout->num_pos();
+                const auto  num_crossings  = gate_level_layout->num_crossings();
+                std::string cost_objective = "A";
+                if (cost == 1)
+                {
+                    cost_objective = "|W|";
+                }
+                else if (cost == 2)
+                {
+                    cost_objective = "|C|";
+                }
+                else if (cost == 3)
+                {
+                    cost_objective = "ACP";
+                }
```

Comment:
Regarding the above suggestion, you could make each entry a pair with their respective string representation added to it so this part can also be simplified.